scripts/tests/test_phase44d_carry_aware_experimental_3x3_scaling_figure.py

```
@@ -119,6 +119,47 @@ def test_validate_rows_rejects_non_positive_serialized_bytes(self):
+        return rows
+
+    def test_validate_rows_rejects_step_set_missing_frontier(self):
+        rows = self._full_canonical_rows(exclude_steps=(1024,))
+        with self.assertRaises(SystemExit) as ctx:
+            MODULE.validate_rows(rows, source=Path("sample.tsv"))
+        self.assertIn("unexpected step counts", str(ctx.exception))
+        self.assertIn("1024", str(ctx.exception))
+
+    def test_validate_rows_rejects_step_set_missing_512(self):
+        rows = self._full_canonical_rows(exclude_steps=(512,))
+        with self.assertRaises(SystemExit) as ctx:
```

Ruff (0.15.11)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)